avm/res/document-db/mongo-cluster/tests/e2e/max/main.test.bicep

```
@@ -0,0 +1,71 @@
+targetScope = 'subscription'
+
+metadata name = 'Using large parameter set'
+metadata description = 'This instance deploys the module with the maximum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-documentdb-mongoclusters-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ddmcdefmax'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      administratorLogin: 'Admin001'
+      administratorLoginPassword: password
+      nodeCount: 2
+      sku: 'M30'
+      storage: 256
+      createMode: 'Default'
+      highAvailabilityMode: false
+      nodeType: 'Shard'
+      networkAcls: {
+        customRules: [
+          {
+            firewallRuleName: 'allow-1.2.3.4-to-5.6.7.8'
+            endIpAddress: '1.2.3.4'
+            startIpAddress: '5.6.7.8'
```

```suggestion
            startIpAddress: '1.2.3.4'
            endIpAddress: '5.6.7.8'
```

I guess it didn't like that (given the error in the workflow 😄 )